python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -978,26 +1008,39 @@ def _filter_by_project(self, actions, sw_entities, project):
             # command did not provide that property, and as such is not a
             # launcher. Similarly, if it's set to None then the same applies
             # and we don't need to test this action for filtering purposes.
-            if action.get("engine_name") is None:
-                continue
-
-            # We're only interested in entities that are referring to the
-            # same engine as is recorded in the action dict.
-            associated_sw = [s for s in sw_entities if s["engine"] == action["engine_name"]]
-
-            # Check the project against the projects list for matching Software
-            # entities. If a Software entity's projects list is empty, then there
-            # is no filtering to be done, as it's accepted by all projects.
-            for sw in associated_sw:
-                for sw_project in sw.get("projects", []):
-                    if sw_project["id"] != project["id"]:
-                        logger.debug("Action %s filtered out due to SW entity projects.", action)
-                        filtered.append(action)
-                        break
-                if action in filtered:
-                    break
+            if "engine_name" not in action:
```

Comment:
I flipped the logic of this loop because we could turn a triple nested loop into a single loop. :)

Reply:
👍